ScoutSuite/providers/aws/facade/ec2.py

```diff
@@ -32,11 +32,19 @@ def __init__(self, session: boto3.session.Session, owner_id: str):
             if 'Value' not in user_data_response['UserData'].keys():
                 return None
             else:
-                value = base64.b64decode(user_data_response['UserData']['Value'])
-                if value[0:2] == b'\x1f\x8b':  # GZIP magic number
-                    return zlib.decompress(value, zlib.MAX_WBITS | 32).decode('utf-8')
-                else:
-                    return value.decode('utf-8')
+                try:
+                    return await self._decode_user_data(user_data_response['UserData']['Value'])
+                except base64.binascii.Error as e:
+                    return await self._decode_user_data(base64.b64decode(user_data_response['UserData']['Value'] + "==="))
+                except Exception as e:
+                    print_exception('Unable to decode EC2 instance user data: {}'.format(e))
+
+    async def _decode_user_data(self, user_data):
```

Comment:
@paurisa how does this look? I've moved the decoding to a specific function, and added a retry if we catch an error.

Reply:
That sounds great!